### PR TITLE
Use go idiomatic to replace loop

### PR DESCRIPTION
**Pull request by tedyu**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Certain loops can be simplified with go idiomatic.

**Special notes for your reviewer**:

```release-note
NONE
```

### CODE REVIEW

**Review by yagonobre — COMMENTED** [review]
Thanks @tedyu 
/ok-to-test
/priority cleanup
/lgtm

**Comment by tedyu** [comment]
@yagonobre 
Can you take another look ?

thanks

**Comment by yagonobre** [comment]
@tedyu SGTM, can you squash your commits?
/priority backlog
/remove-sig api-machinery

**Comment by tedyu** [comment]
ping @jpbetz

File: cluster/images/etcd/migrate/rollback_v2.go [hunk 1]
```diff
@@ -185,9 +185,7 @@ func traverseMetadata(head *store.NodeExtern, handleFunc func(*store.NodeExtern)
 
 		handleFunc(n)
 
-		for _, next := range n.Nodes {
```
Comment (yue9944882) [inline cluster/images/etcd/migrate/rollback_v2.go]: not only styling but it's also improving performance codewise (well, a bit)

/lgtm

**Comment by bart0sh** [comment]
/lgtm

@kubernetes/sig-cluster-lifecycle-pr-reviews please, approve

**Comment by tedyu** [comment]
ping @wenjiaswe @jpbetz

**Comment by tedyu** [comment]
ping @wenjiaswe @jpbetz

**Comment by wenjiaswe** [comment]
Thanks @tedyu 
/lgtm

**Comment by tedyu** [comment]
ping @wenjiaswe

**Comment by tedyu** [comment]
@jpbetz @wenjiaswe 
All the tests passed.

**Comment by jpbetz** [comment]
Thanks @tedyu it will automerge once prow completes pre-merge testing, you can track status on https://prow.k8s.io/pr?query=is%3Apr%20state%3Aopen%20author%3Atedyu

**Pull request merged**